pkg/operator/ceph/object/spec.go

```diff
@@ -540,8 +558,34 @@ func (c *clusterConfig) generateVolumeSourceWithTLSSecret() (*v1.SecretVolumeSou
 	return secretVolSrc, nil
 }
 
-func (c *clusterConfig) rgwTLSSecretType() (v1.SecretType, error) {
-	rgwTlsSecret, err := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(context.TODO(), c.store.Spec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+func (c *clusterConfig) generateVolumeSourceWithCaBundleSecret() (*v1.SecretVolumeSource, error) {
```

Comment:
It doesn't look like you need to return an error. The only error condition below should not happen since the caller already checked for `CaBundleRef != ""`.
```suggestion
func (c *clusterConfig) generateVolumeSourceWithCaBundleSecret() *v1.SecretVolumeSource {
```

Reply:
Alright, since we are agree that secret's type doesn't matter.

Reply:
This won't work for k8s tls secret type, because it has predefined data fields like `tls.cert` and `tls.key`. IMO only opaque type users can have customed data fields (I might be wrong here)

Reply:
that was initially mention in 1.6 docs, https://rook.io/docs/rook/v1.6/ceph-object-store-crd.html#gateway-settings - but in master no such notes

Reply:
https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets also says that this type should contains `tls.key` and `tls.cert`, but no any note that other fields are not acceptable. But from cli help command it seems that only those 2 keys can be passed:

`kubectl create secret tls -h
Create a TLS secret from the given public/private key pair.

 The public/private key pair must exist before hand. The public key certificate must be .PEM encoded and match the given
private key.

Examples:
  kubectl create secret tls tls-secret --cert=path/to/tls.cert --key=path/to/tls.key

Options:
      --allow-missing-template-keys=true: If true, ignore any errors in templates when a field or map key is missing in
the template. Only applies to golang and jsonpath output formats.
      --append-hash=false: Append a hash of the secret to its name.
      --cert='': Path to PEM encoded public key certificate.
      --dry-run='none': Must be "none", "server", or "client". If client strategy, only print the object that would be
sent, without sending it. If server strategy, submit server-side request without persisting the resource.
      --key='': Path to private key associated with given certificate.
  -o, --output='': Output format. One of:
json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-file.
      --save-config=false: If true, the configuration of current object will be saved in its annotation. Otherwise, the
annotation will be unchanged. This flag is useful when you want to perform kubectl apply on this object in the future.
      --template='': Template string or path to template file to use when -o=go-template, -o=go-template-file. The
template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].
      --validate=true: If true, use a schema to validate the input before sending it

Usage:
  kubectl create secret tls NAME --cert=path/to/cert/file --key=path/to/key/file [--dry-run=server|client|none]
[options]

Use "kubectl options" for a list of global command-line options (applies to all commands).`

So it looks like, we don't have any other option - only opaque.